samples/application_development/external_lib/CMakeLists.txt

```diff
@@ -1,3 +1,4 @@
+cmake_minimum_required(VERSION 2.4)
```

Comment:
As all the samples seem to include boilerplate.cmake then why not add this cmake_minimum_required() there? This way this setting is always applied even for new samples and it is easy to set the required version in just one place.

Reply:
Because CMake does not allow this.

"a call in an included file is not sufficient"

https://cmake.org/cmake/help/v3.12/policy/CMP0000.html

Reply:
Ok, perhaps that could be documented in the commit message. Now one starts to wonder why this change is needed unless one knows CMake well enough.

Reply:
Would be nice to have it in the docs somewhere too. Looks cryptic otherwise.

Reply:
Added a clarifying CMake comment.